shell/browser/net/asar/asar_file_validator.h

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2021 Slack Technologies, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef SHELL_BROWSER_NET_ASAR_ASAR_FILE_VALIDATOR_H_
+#define SHELL_BROWSER_NET_ASAR_ASAR_FILE_VALIDATOR_H_
+
+#include <algorithm>
+#include <memory>
+
+#include "crypto/secure_hash.h"
```

Comment:
This can't be forward declared because the destructor needs size information of `SecureHash`.  E.g. `invalid application of 'sizeof' to an incomplete type 'crypto::SecureHash'`

Reply:
hm, that's surprising, because the only reference to `crypto::SecureHash` in the header file is in `unique_ptr`, so the size information should always be the size of a pointer. Elsewhere we are able to forward-declare things that we only refer to in the template parameter of `unique_ptr` i'm pretty sure.